td-shim/src/bin/td-shim/main.rs

```diff
@@ -540,3 +511,32 @@ fn prepare_hob_list(
 
     hob_base
 }
+
+fn verify_payload<'a>(payload: &'a [u8], td_event_log: &mut TdEventLog) -> &'a [u8] {
+    #[cfg(feature = "secure-boot")]
+    {
+        let cfv = memslice::get_mem_slice(memslice::SliceType::Config);
+        let verifier = verifier::PayloadVerifier::new(payload, cfv);
```

Comment:
using `expect!("Secure Boot: Cannot read verify header from payload binary");` may simplify the code here.